--- json_file_module.F90	2023-02-16 13:07:18.019222400 +0900
+++ json_file_module.F90	2023-02-16 13:09:29.969222400 +0900
@@ -1013,9 +1013,21 @@
     integer(IK),intent(out),optional    :: var_type   !! variable type
     integer(IK),intent(out),optional    :: n_children !! number of children
     character(kind=CK,len=:),allocatable,intent(out),optional :: name !! variable name
+#if defined __GFORTRAN__
+    character(kind=CK,len=:),allocatable :: p_name  !! temporary variable for getting name
+#endif
 
+#if defined __GFORTRAN__
+    if (present(name)) then !workaround for gfortran bug
+        call me%core%info(me%p,path,found,var_type,n_children,p_name)
+        name = p_name
+    else
+        call me%core%info(me%p,path,found,var_type,n_children)
+    end if
+#else
     call me%core%info(me%p,path,found,var_type,n_children,name)
-
+#endif
+    
     end subroutine json_file_variable_info
 !*****************************************************************************************
 
